### PR TITLE
WAF: Avoid using Waf_Rules_Manager from Waf_Runner::initialize()

### PR DESCRIPTION
**Pull request by nateweller**
Related to https://github.com/Automattic/jpop-issues/issues/9175

## Proposed changes:

* Migrates `Waf_Rules_Manager::RULES_ENTRYPOINT_FILE` to a new `JETPACK_WAF_ENTRYPOINT` constant, to avoid autoloading the `Waf_Rules_Manager` class during standalone mode firewall execution.
* Deprecates the `Waf_Rules_Manager::RULES_ENTRYPOINT_FILE` constant.

### Other information:

- [ ] Have you written new tests for your changes, if applicable?
- [x] Have you checked the E2E test CI results, and verified that your changes do not break them?
- [ ] Have you tested your changes on WordPress.com, if applicable (if so, you'll see a generated comment below with a script to run)?

## Jetpack product discussion

peb6dq-2HL-p2

## Does this pull request change what data or activity we track or use?

No

## Testing instructions:

* Set up automatic firewall rules.
* Verify the firewall blocks a request in both the default and standalone mode.
* Test updating various firewall settings.

### CODE REVIEW

**Comment by nateweller** [comment]
PHP compatibility warnings appear unrelated to the changes in this PR.

File: projects/packages/waf/src/class-waf-rules-manager.php [hunk 1]
```diff
@@ -39,10 +39,9 @@ class Waf_Rules_Manager {
 	const IP_LISTS_ENABLED_OPTION_NAME = 'jetpack_waf_ip_list';
 
 	// Rule Files
-	const RULES_ENTRYPOINT_FILE = '/rules/rules.php';
```
Comment (dkmyta) [inline projects/packages/waf/src/class-waf-rules-manager.php:42]: This constant is still used in the Debug Helper plugin on line 226 in `class-waf-helper.php`.

**Review by dkmyta — CHANGES_REQUESTED** [review]

File: projects/packages/waf/src/class-waf-rules-manager.php (hunk 1 above)
Comment (nateweller) [reply projects/packages/waf/src/class-waf-rules-manager.php:42]: Good catch! Fixed with [5782afb](https://github.com/Automattic/jetpack/pull/38944/commits/5782afb930bb92664d9cee07d19ef376058d1a04).

File: projects/packages/waf/src/class-waf-rules-manager.php [hunk 2]
```diff
@@ -223,15 +222,15 @@ public static function generate_rules() {
 		Waf_Runner::initialize_filesystem();
 
 		$rules                = "<?php\n";
-		$entrypoint_file_path = Waf_Runner::get_waf_file_path( self::RULES_ENTRYPOINT_FILE );
+		$entrypoint_file_path = Waf_Runner::get_waf_file_path( Waf_Runner::ENTRYPOINT_FILE );
```
Comment (nateweller) [inline projects/packages/waf/src/class-waf-rules-manager.php]: I am wondering if this is problematic - will this `generate_rules` method run within a request where an outdated `Waf_Runner` class was autoloaded in standalone mode, creating the same issue?

Wondering if we just need to duplicate the deceleration of the entrypoint file value, or provide it from a third source separate from the runner and rule manager classes... or just safely handle it potentially being undefined.

🤔

**Review by dkmyta — APPROVED** [review]
Tests well!

**Pull request merged**